include/task.h

```diff
@@ -68,6 +68,33 @@ namespace detail {
 		std::unordered_multimap<buffer_id, std::unique_ptr<range_mapper_base>> map;
 	};
 
+	class side_effect_map : private std::unordered_map<host_object_id, experimental::side_effect_order> {
+	  private:
+		using map_base = std::unordered_map<host_object_id, experimental::side_effect_order>;
+
+	  public:
+		using typename map_base::const_iterator, map_base::value_type, map_base::key_type, map_base::mapped_type, map_base::const_reference,
+		    map_base::const_pointer;
+		using iterator = const_iterator;
+		using reference = const_reference;
+		using pointer = const_pointer;
+
+		using map_base::size, map_base::count, map_base::empty, map_base::cbegin, map_base::cend, map_base::at;
+
+		iterator begin() const { return cbegin(); }
+		iterator end() const { return cend(); }
+		iterator find(host_object_id key) const { return map_base::find(key); }
+
+		void add_side_effect(host_object_id hoid, experimental::side_effect_order order);
+	};
+
+	struct task_geometry {
+		int dimensions = 0;
+		cl::sycl::range<3> global_size{1, 1, 1};
```

Comment:
Should this default to `{1,1,1}` or `{0,0,0}`? I think the current implementation might be a change in behaviour.

Reply:
You're right! Fixed.